mycpp/gc_list.h

```diff
@@ -342,7 +342,7 @@ T List<T>::pop(int i) {
   len_--;
 
   // Shift everything by one
-  memmove(slab_->items_ + i, slab_->items_ + (i + 1), len_ * sizeof(T));
+  memmove(slab_->items_ + i, slab_->items_ + (i + 1), (len_ - i) * sizeof(T));
```

Comment:
We might want to cherry-pick this out of here. I ran into this issue as described in the commit https://github.com/oils-for-unix/oils/pull/2028/commits/2bc8a856fec952c8a4522b7d824e8da4f301689f. But this is not necessary for the current form of this PR.

Reply:
Oh thanks for finding this!   

I wouldn't mind if it was a separate PR with a unit test --  I guess we are missing some coverage

Reply:
Okay, I'll revert it here and send up another PR.